core/src/main/java/io/grpc/MethodDescriptor.java

```diff
@@ -262,4 +269,40 @@ public static String extractFullServiceName(String fullMethodName) {
     }
     return fullMethodName.substring(0, index);
   }
+
+  /**
+   * An internal only class for passing information directly into the transport.
+   */
+  @Internal
+  public static final class InternalAccessor {
+    private static final String PACKAGE = InternalAccessor.class.getPackage().getName();
+
+    private static final Set<String> PACKAGE_WHITELIST =
+        new HashSet<String>(Arrays.asList(PACKAGE + ".netty."));
+
+    /**
+     * Only packages on the whitelist may access this.
+     */
+    @Internal
+    public InternalAccessor() {
+      StackTraceElement[] st = new Throwable().getStackTrace();
```

Comment:
Also, I realize this is not exactly clean, but its easy to remove later.  Would you consider letting this in, under the premise that it can be removed if it causes problems later?